src/App.tsx

```diff
@@ -462,6 +488,45 @@ function App() {
     setInvokeMethodResults({});
   };
 
+  /**
+   * setup caip294:wallet_announce event listener
+   * docs: https://github.com/ChainAgnostic/CAIPs/blob/bc4942857a8e04593ed92f7dc66653577a1c4435/CAIPs/caip-294.md#specification
+   */
+  useEffect(() => {
+    const handleWalletAnnounce = (ev: Event) => {
+      const customEvent = ev as CustomEvent;
+      const _extensionId = customEvent.detail.params.extensionId ?? '';
```

Comment:
can we rename this to something like `newExtensionId` maybe?

Reply:
resolved in [de81e](https://github.com/MetaMask/test-dapp-multichain/pull/22/commits/de81e7e392022e41ac8edd4066a44826540fb4d5)